src/components/Backdrop/index.js

```diff
@@ -53,10 +75,20 @@ export default class Backdrop extends Component {
   };
 
   render() {
-    const { children, overlay, overlayProps } = this.props;
+    const { children, overlay, overlayProps, isVisible, animated } = this.props;
+
+    if (isVisible && !this.hasListeners) {
+      this.attachListeners();
+    } else if (!isVisible && this.hasListeners) {
+      this.detachListeners();
+    }
```

Comment:
Is `componentDidMount` and `componentDidUpdate` a better place for this logic? I'd like to keep this logic out of the `render` method, if possible.

Reply:
Yes, I was thinking about it, unfortunately **componentDidUpdate()** is not called for the initial render, so one way or another we do need to call it from there.

Reply:
But since `componentDidMount` is, can you please implement this logic there and in `componentDidUpdate` instead of `render`?

Reply:
Done.